src/error-table.rkt

```diff
@@ -0,0 +1,49 @@
+#lang racket
+
+(require "points.rkt" "syntax/types.rkt" "core/localize.rkt" "timeline.rkt")
+
+(provide group-errors)
+
+(define (zip l1 l2) (map list l1 l2))
+
+(define (group-errors expr pcontext)
+  (define points (for/list ([(pt _) (in-pcontext pcontext)]) pt))
+  (define local-errors (car (compute-local-errors (list expr) (*context*))))
+
+  (define max-per-point (make-hash))
+  (define errpts-per-subexpr (make-hash))
+
+  (for ([pt points])
+    (hash-set! max-per-point pt (list 0 0)))
+
+  (for ([(sub-expr errors) local-errors])
+    (hash-set! errpts-per-subexpr sub-expr null)
+    (for ([pair (zip points errors)])
+      (define pt (car pair))
+      (define err (cadr pair))
+      (define old-error (hash-ref max-per-point pt))
+      (if (> err (cadr old-error))
+        (hash-set! max-per-point pt (list sub-expr err))
+        null)))
+  
+  (hash-set! errpts-per-subexpr #f null)
+
+  (for ([(k v) max-per-point])
+    (define sub-expr (car v))
+    (define err (cadr v))
+    (if (> err 2)
+      (hash-update! errpts-per-subexpr sub-expr (curry cons k))
+      (hash-update! errpts-per-subexpr #f (curry cons k))))
+  
+  ; (eprintf "[grouped errors]\n")
+  ; (for ([(k v) errpts-per-subexpr])
+  ;   (eprintf "~a, ~a \n" k v))
+  
+  ; (eprintf "\n[percentage-errored]\n")
+  (for ([(k v) errpts-per-subexpr])
+    (define num-of-errors (length v))
+    ; (eprintf "~a : ~a\n" k num-of-errors)
+
+    (if (= 0 num-of-errors)
+      #f
```

Comment:
A few Racket notes here:

1. If you want a "do-nothing" expression the recommended one is `(void)`, not `#f`
2. If you want an if statement where one side does nothing, use `when` and `unless`
3. Instead of checking if the length of something is zero, just use `(null? v)`
4. If you have a `for` loop where you only want to execute some iterations, just use `#:when` in the `for` loop (check the manual).

So the code could just be:

```
(for ([(k v) (in-hash errpts-per-subexpr)] #:unless (null? v)])
  (timeline-push! 'problems (~a k) (length v)))
```

Way shorter and clearer.